### ISSUE TITLE
vine: unable to pack environment using poncho_package_create

### ISSUE THREAD

**Issue by JinZhou5042** [cooperative-computing-lab/cctools#3862]
I just downloaded the ndcctools package using conda
```
conda install ndcctools
```
Then used `poncho_package_create` to pack my current python environment
```
poncho_package_create $CONDA_PREFIX coffea2024.tar.gz
```
Then tried to launch a worker using `vine_factory` with the following command
```
vine_factory -T condor -C factory.json --scratch-dir /scratch365/jzhou24 --python-env coffea2024.tar.
gz
```
and json file
```
{
    "manager-name": "jzhou24-hgg",
    "max-workers": 1,
    "min-workers": 1,
    "workers-per-cycle": 1,
    "cores": 12,
    "memory": 200000,
    "disk": 200000,
    "condor_requirements": "((has_vast))",
    "timeout": 36000
}
```
However, I got an error message showing that the factory is unable to copy the tarball into my scratch directory

```
(coffea2024) [jzhou24@crcfe02 jzhou24]$ vine_factory -T condor -C factory.json --scratch-dir /scratch365/jzhou24 --python-env coffea2024.tar.gz
Configuration file '/project01/ndcms/jzhou24/factory.json' has been loaded.manager-name: jzhou24-hgg
max-workers: 1
min-workers: 1
workers-per-cycle: 1
tasks-per-worker: 12
timeout: 36000 s
cores: 12.000
memory: 200000 MB
disk: 200000 MB

vine_factory: Cannot copy wrapper input file coffea2024.tar.gz to factory scratch directory /scratch365/jzhou24/coffea2024.tar.gz:
Invalid argument
```

I am pretty sure the `coffea2024.tar.gz` had been copied to the target place as expected. I reinstalled the environment twice but didn't help.

I used to be able to add environment with the above procedure, is there something missing?

**Comment by dthain** [comment]
I don't think we have any special insight into this problem, so you will have to debug it.
I would start with the given error message:
```
vine_factory: Cannot copy wrapper input file coffea2024.tar.gz to factory scratch directory /scratch365/jzhou24/coffea2024.tar.gz:
```

And then start digging in from top to bottom:
- Do those files/directories exist?
- Can you read and write them?
- What is the function that does the file copy? (Start with `vine_factory.c`)
- Where does it fail?  (Use strace or gdb to find out.)

**Comment by dthain** [comment]
Ah, this could be a relevant fact:  the environment file is quite large: 3.7GB.  I have a suspicion that number is overflowing a 32-bit signed integer somewhere:
```
-rw-------   1 jzhou24 campus 3793424253 Jun 13 18:06 coffea2024.tar.gz
```

**Comment by BarrySlyDelgado** [comment]
This may not be the source of the error, but I once had a similar message when running the manager from the same location of the factory's scratch directory.

**Comment by btovar** [comment]
That might be it. The `cp` fails because source and destination are the same file.

**Comment by JinZhou5042** [comment]
@dthain You are right, the return type of `copy_file_to_file` is `int64_t` on success, but the variable receiving this value is `int`. I modified the relevant variables to `int64_t` and the errors disappeared.

**Comment by JinZhou5042** [comment]
addressed in #3863
